packages/starknet-snap/src/index.test.ts

```diff
@@ -50,15 +51,30 @@ describe('onRpcRequest', () => {
     expect(createAccountSpy).toHaveBeenCalledTimes(1);
   });
 
-  it('throws `Unable to execute the rpc request` error if an error has thrown', async () => {
-    const { createAccountSpy, keyPairSpy } = createMockSpy();
+  it('throws `MethodNotFoundError` if the request method not found', async () => {
+    await expect(
+      onRpcRequest({
+        ...createMockRequest(),
+        request: {
+          ...createMockRequest().request,
+          method: 'method_not_found',
+        },
+      }),
+    ).rejects.toThrow(MethodNotFoundError);
+  });
 
-    createAccountSpy.mockRejectedValue(new Error('Custom Error'));
-    keyPairSpy.mockReturnThis();
+  it('throws `SnapError` if the error is an instance of SnapError', async () => {
+    const { createAccountSpy } = createMockSpy();
+    createAccountSpy.mockRejectedValue(new SnapError('error'));
 
-    await expect(onRpcRequest(createMockRequest())).rejects.toThrow(
-      'Unable to execute the rpc request',
-    );
+    await expect(onRpcRequest(createMockRequest())).rejects.toThrow(SnapError);
+  });
+
+  it('throws `SnapError` if the error is not an instance of SnapError', async () => {
```

Comment:
This wording is a bit misleading. Does this mean we always throw SnapError ?

Reply:
Maybe we should link it with the log level ?

Reply:
yes we always throw SnapError , it has nothing to do with log level